### ISSUE TITLE
[regression/8.0.3] Border inside a RefreshView ruins the width of other controls on iOS

### ISSUE THREAD

**Issue by haavamoa** [dotnet/maui#18852]
### Description

When using a `Border`  inside a `RefreshView`+`ScrollView` (which is normal when you want pull to refresh for a page), it ruins its width and all other controls width inside the ScrollView:

```xml
<RefreshView>
    <ScrollView>
        <VerticalStackLayout BackgroundColor="Green"
                             Padding="5">
            <Label Text="Test1"
                   Background="Red" />
            <Border Background="Blue" >
                <Label Text="Title" />
            </Border>
        </VerticalStackLayout>
    </ScrollView>
</RefreshView>
```
**This is illustrated by looking at the width of `Label` (red) and `VerticalStackLayout` (green) in this example:**

<img width="488" alt="image" src="https://github.com/dotnet/maui/assets/2527084/cbdb5335-9af2-4ca6-ab18-70d4d0796fd7">

**Once I remove the `Border`, everything works as expected:**

<img width="488" alt="image" src="https://github.com/dotnet/maui/assets/2527084/b5731082-f183-4c15-b749-14d4088d3456">

**Once I build the same example on .NET 7.0 it works as expected:**
<img width="488" alt="image" src="https://github.com/dotnet/maui/assets/2527084/567acda4-8d8d-4125-be57-cf6a6aa6065f">

### Steps to Reproduce

1. [Clone this branch in this repository.](https://github.com/haavamoa/MauiApp/tree/issue/border-refreshview-dotnet8)
2. Start the app
3. Observe the green, red and blue controls not filling the entire width of the screen.

### Link to public reproduction project repository

https://github.com/haavamoa/MauiApp/tree/issue/border-refreshview-dotnet8

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

7.0.101

### Affected platforms

iOS

### Affected platform versions

Tested on iOS 17

### Did you find any workaround?

A rather hacky work around is to subscribe to `SizeChanged` on the `VerticalStackLayout` and explicitly set the `WidthRequest` to the `Parent.Parent.Width`. This will make the width fill the width of the `RefreshView`. Not very nice, but it works.

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.9.0 Preview 1(8.0.3). Repro on iOS 17.0, not repro on Windows 11, Android 14.0-API34 and MacCatalyst with below Project:
[18852.zip](https://github.com/dotnet/maui/files/13612169/18852.zip)

**Comment by kubaflo** [comment]
This seems to be fixed  on `8.0.10-nightly.10044`
<img src="https://github.com/dotnet/maui/assets/42434498/2ad963b4-5ecf-4807-bc19-fb273d53522d" width="250px"/>
